src/Shared/Recruit.Vacancies.Client/Infrastructure/Services/ProviderRelationship/ProviderRelationshipsService.cs

```diff
@@ -32,7 +32,39 @@ public class ProviderRelationshipsService : IProviderRelationshipsService
 
         public async Task<IEnumerable<EmployerInfo>> GetLegalEntitiesForProviderAsync(long ukprn)
         {
-            ProviderPermissions providerPermissions = null;
+            var providerPermissions = await GetProviderPermissionsAsync(ukprn);
+
+            if (providerPermissions == null)
+                return Array.Empty<EmployerInfo>();
+
+            return await GetEmployerInfosAsync(providerPermissions);
+        }
+
+        public async Task<bool> ProviderHasPermissionAsync(long ukprn, string accountPublicHashedId, long legalEntityId)
+        {
+            var providerPermissions = await GetProviderPermissionsAsync(ukprn);
+
+            var permittedLegalEntities = providerPermissions?.AccountProviderLegalEntities
+                .Where(l => l.AccountPublicHashedId == accountPublicHashedId)
+                .ToList();
+
+            if (permittedLegalEntities == null || permittedLegalEntities.Any() == false)
```

Comment:
(see my comment on `GetProviderPermissionsAsync` first) I think if `GetProviderPermissionsAsync` is updated to return empty list, then the check for `.Any()` can be moved up before populating `permittedLegalEntities`.

Reply:
I can remove the null check but I still need to keep this check here as the permissions get filtered to the employer account so could well be empty at this point.